packages/cli/src/execution-lifecycle/execution-lifecycle-hooks.ts

```diff
@@ -502,27 +484,19 @@ export function getWorkflowHooksWorkerExecuter(
 	mode: WorkflowExecuteMode,
 	executionId: string,
 	workflowData: IWorkflowBase,
-	optionalParameters?: IWorkflowHooksOptionalParameters,
+	optionalParameters: IWorkflowHooksOptionalParameters = {},
 ): WorkflowHooks {
-	optionalParameters = optionalParameters || {};
-	const hookFunctions = hookFunctionsSaveWorker();
-	const preExecuteFunctions = hookFunctionsPreExecute();
-	for (const key of Object.keys(preExecuteFunctions)) {
-		const hooks = hookFunctions[key] ?? [];
-		hooks.push.apply(hookFunctions[key], preExecuteFunctions[key]);
-	}
+	const toMerge = [
+		hookFunctionsNodeTelemetry(),
+		hookFunctionsSaveWorker(),
+		hookFunctionsPreExecute(),
+	];
 
 	if (mode === 'manual' && Container.get(InstanceSettings).isWorker) {
-		const pushHooks = hookFunctionsPush();
-		for (const key of Object.keys(pushHooks)) {
-			if (hookFunctions[key] === undefined) {
-				hookFunctions[key] = [];
-			}
-			// eslint-disable-next-line prefer-spread
-			hookFunctions[key].push.apply(hookFunctions[key], pushHooks[key]);
-		}
+		toMerge.push(hookFunctionsPush());
 	}
 
+	const hookFunctions = mergeHookFunctions(...toMerge);
```

Comment:
`mergeHookFunctions` should accept an array or a variadic argument?

Reply:
I'm fine either way. do you prefer one over the other?

Reply:
Sorry, I should've been clearer. I meant it should accept a union of them, so cases like here can be passed directly, and cases like above where we define hooks inline work as well. Super minor, feel free to ignore.

Reply:
Like this: https://github.com/n8n-io/n8n/blob/5f1adefca71a8f5364ebd1e0ce602db0929de319/packages/cli/src/services/role.service.ts#L112-L115

Reply:
I'm not sure if this specific code will stay after the next few refactors. maybe we can leave this as it is for now?